CesiumVectorData/include/CesiumVectorData/VectorRasterizer.h

```diff
@@ -0,0 +1,120 @@
+#pragma once
+
+#include "VectorStyle.h"
+
+#include <CesiumGeometry/Rectangle.h>
+#include <CesiumGeospatial/CartographicPolygon.h>
+#include <CesiumGeospatial/Ellipsoid.h>
+#include <CesiumGeospatial/GlobeRectangle.h>
+#include <CesiumGltf/ImageAsset.h>
+#include <CesiumUtility/Color.h>
+#include <CesiumUtility/IntrusivePointer.h>
+#include <CesiumUtility/ReferenceCounted.h>
+#include <CesiumVectorData/GeoJsonObject.h>
+
+#include <blend2d.h>
+#include <blend2d/context.h>
+#include <blend2d/image.h>
+
+#include <span>
+
+namespace CesiumVectorData {
+
+/**
+ * @brief Rasterizes vector primitives into a \ref CesiumGltf::ImageAsset.
+ */
+class VectorRasterizer {
+public:
+  /**
+   * @brief Creates a new \ref VectorRasterizer representing the given rectangle
+   * on the globe.
+   *
+   * @param bounds The bounds on the globe that this rasterizer's canvas will
+   * cover.
+   * @param imageAsset The destination image asset. This \ref
+   * CesiumGltf::ImageAsset must be four channels, with
+   * only one byte per channel (RGBA32).
+   * @param mipLevel The mip level that the rasterizer should rasterize for the
+   * image.
+   * @param ellipsoid The ellipsoid to use.
+   */
+  VectorRasterizer(
+      const CesiumGeospatial::GlobeRectangle& bounds,
+      CesiumUtility::IntrusivePointer<CesiumGltf::ImageAsset>& imageAsset,
+      uint32_t mipLevel = 0,
+      const CesiumGeospatial::Ellipsoid& ellipsoid =
+          CesiumGeospatial::Ellipsoid::WGS84);
+
+  /**
+   * @brief Draws a \ref CesiumGeospatial::CartographicPolygon to the canvas.
+   *
+   * @param polygon The polygon to draw.
+   * @param style The \ref VectorStyle to use when drawing the polygon.
+   */
+  void drawPolygon(
+      const CesiumGeospatial::CartographicPolygon& polygon,
+      const VectorStyle& style);
```

Comment:
If you take my previous suggestion for changes to `PolygonStyle`, then I think this method could simply take a `PolygonStyle` instead of an entire `VectorStyle`. Similar with the other methods; they can just take the style type appropriate for what they're drawing.